### PR TITLE
FFC-171848: Defra Id Register your interest

### PR DESCRIPTION
**Pull request by json-hamilton**
https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/171848

### CODE REVIEW

File: app/routes/register-your-interest/register-your-interest.js [hunk 1]
```diff
@@ -8,7 +9,11 @@ module.exports = [
     options: {
       auth: false,
       handler: async (request, h) => {
-        return h.view('register-your-interest/register-your-interest', { ruralPaymentsAgency })
+        if (defraIdConfig.enabled) {
+          return h.view('defra-id/register-your-interest/register-your-interest', { ruralPaymentsAgency })
```
Comment (liamwilson-kainos) [inline app/routes/register-your-interest/register-your-interest.js:13]: could probably remove the duplicate registration of interest and have this as an index page?

File: app/views/defra-id/register-your-interest/register-your-interest.njk [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }}
+
+    <h2 class="govuk-heading-m">Help with registering</h2>
+
+    <p>If you need to update your main business email address, <a href="https://www.ruralpayments.service.gov.uk/">sign in to the Rural Payments service.</a></p>
+    <p class="govuk-heading-body">If you have another question about your registration, contact the RPA by email or phone:</p>
+
+    {{ rpaContactDetails({email: ruralPaymentsAgency.email, callChargesUri: ruralPaymentsAgency.callChargesUri, telephone: ruralPaymentsAgency.telephone}) }}
+    
+    <p>If you do not have an account, <a href="https://www.ruralpayments.service.gov.uk/">register with the Rural Payments service.</a></p>
```
Comment (liamwilson-kainos) [inline app/views/defra-id/register-your-interest/register-your-interest.njk:62]: rural payment service url should also be coming from the config and we can pass a value for it into this page

**Review by liamwilson-kainos — CHANGES_REQUESTED** [review]

File: app/routes/register-your-interest/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+              'string.base': ERROR_MESSAGE.enterYourEmailAddress,
+              'string.empty': ERROR_MESSAGE.enterYourEmailAddress,
+              'string.email': ERROR_MESSAGE.enterYourEmailAddressInCorrectFormat
+            })
+        }),
+        failAction: async (request, h, error) => {
+          const errorMessages = error
+            .details
+            .reduce((acc, e) => ({ ...acc, [e.context.label]: { text: e.message } }), {})
+          if (defraIdConfig.enabled === true) {
```
Comment (liamwilson-kainos) [inline app/routes/register-your-interest/index.js:46]: not sure if we need this check seeing that we will only ever POST to `register-your-interest` once DEFRA ID is enabled. Also we need to make sure when we submit the form we are sending the message to the queue - .e.g.

`await sendRegisterYourInterestMessage(sbi, crn, emailAddress)`

File: app/routes/register-your-interest/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            {
+              ...request.payload,
+              ruralPaymentsAgency,
+              errorMessages
+            }
+          ).code(400).takeover()
+        }
+      },
+      handler: async (request, h) => {
+        if (defraIdConfig.enabled === true) {
```
Comment (liamwilson-kainos) [inline app/routes/register-your-interest/index.js:68]: not sure we even need the enabled check as when turned off i don't think we ever POST to `register-your-interest`

**Review by liamwilson-kainos — CHANGES_REQUESTED** [review]

File: app/routes/register-your-interest/index.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+              ...request.payload,
+              ruralPaymentsAgency,
+              errorMessages
+            }
+          ).code(400).takeover()
+        }
+      }
+    },
+    handler: async (request, h) => {
+      await sendEmail(registerYourInterest, request.payload.emailAddress)
```
Comment (liamwilson-kainos) [inline app/routes/register-your-interest/index.js:61]: We can use this new template which is a registration of interest confirmation V2 (new for defra id)-> 

https://www.notifications.service.gov.uk/services/4e30070f-25e2-40df-8add-e4375408638f/templates/8c6d9dc1-127d-4355-a16c-cc5fd522f64f

ID is 8c6d9dc1-127d-4355-a16c-cc5fd522f64f

File: app/views/defra-id/register-your-interest/registration-complete.njk [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{% extends './layouts/layout.njk' %}
+
+{% block pageTitle %}{{serviceName}}{% endblock %}
+
+{% block content %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      {{ govukPanel({
+        titleText: "Registration complete - DRAFT PAGE"
```
Comment (liamwilson-kainos) [inline app/views/defra-id/register-your-interest/registration-complete.njk:9]: we could replace the placeholder content with https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/170010/ and form the prototype  https://fcp-ahwr-prototype.herokuapp.com/v10/registration/confirmation

**Review by liamwilson-kainos — DISMISSED** [review]

File: app/views/defra-id/register-your-interest/index.njk [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        },
+        classes: "govuk-input govuk-!-width-three-quarters",
+        id: "emailAddress",
+        name: "emailAddress",
+        value: emailAddress,
+        errorMessage: errorMessages.emailAddress
+      }) 
+    }}
+
+    <p class="govuk-heading-body">The Rural Payments Agency will contact you with a link to start your application as soon as possible.</p>
```
Comment (chrissmKainos) [inline app/views/defra-id/register-your-interest/index.njk:32]: Prototype show '_Rural Payments Agency (RPA)_'

https://fcp-ahwr-prototype.herokuapp.com/v10/registration/register

File: app/views/defra-id/register-your-interest/index.njk [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        isStartButton: true,
+        text: "Register",
+        preventDoubleClick: true
+      }) 
+    }}
+
+    <h2 class="govuk-heading-m">Help with registering</h2>
+
+    <p>If you need to update your main business email address, <a href="{{ ruralPaymentsAgency.loginUri }}">sign in to the Rural Payments service.</a></p>
+    <p class="govuk-body">If you cannot apply online or have a question about your application contact the Rural Payments Agency by email or phone.</p>
```
Comment (chrissmKainos) [inline app/views/defra-id/register-your-interest/index.njk:50]: Different text in the prototype, '_If you cannot apply online or have a question..._' is '_If you have another question about your registration..._'

https://fcp-ahwr-prototype.herokuapp.com/v10/registration/register

**Review by chrissmKainos — DISMISSED** [review]

**Review by marcin-mogiela — DISMISSED** [review]

**Review by liamwilson-kainos — APPROVED** [review]

**Review by liamwilson-kainos — APPROVED** [review]

**Pull request merged**